controller/appcontroller.go

```diff
@@ -254,13 +255,27 @@ func (ctrl *ApplicationController) setAppManagedResources(a *appv1.Application,
 }
 
 // returns true of given resources exist in the namespace by default and not managed by the user
-func isKnownOrphanedResourceExclusion(key kube.ResourceKey) bool {
+func isKnownOrphanedResourceExclusion(key kube.ResourceKey, proj *appv1.AppProject) bool {
 	if key.Namespace == "default" && key.Group == "" && key.Kind == kube.ServiceKind && key.Name == "kubernetes" {
 		return true
 	}
 	if key.Group == "" && key.Kind == kube.ServiceAccountKind && key.Name == "default" {
 		return true
 	}
+
+	list := proj.Spec.OrphanedResourceWhitelist
+	for _, item := range list {
+		ok, err := filepath.Match(item.Kind, key.Kind)
+		if ok && err == nil {
+			ok, err = filepath.Match(item.Group, key.Group)
+			if ok && err == nil {
+				ok, err = filepath.Match(item.Name, key.Name)
+				if ok && err == nil {
+					return true
```

Comment:
@alexmt I think we need to discuss further with the matching logic here if any of the field is optional